src/erldns_resolver.erl

```diff
@@ -148,7 +148,7 @@ resolve_exact_type_match(Message, Qname, Qtype, Host, CnameChain, MatchedRecords
             false ->
               resolve_exact_type_match(Message, Qname, Qtype, Host, CnameChain, MatchedRecords, Zone, _AuthorityRecords, NSRecords)
           end;
-        {error, authority_not_found} ->
+        {error, _} ->
```

Comment:
I understand now why you made the change here. Does it make sense to be explicit with a new case pattern for `{error, no_question}` that has the same logic as `{error, authority_not_found}` for now?